src/ext/coms.c

```diff
@@ -218,46 +219,42 @@ BOOL_T ddtrace_coms_rotate_stack(BOOL_T attempt_allocate_new) {
             *next_stack = new_stack();
             atomic_store(&ddtrace_coms_global_state.current_stack, *next_stack);
             *next_stack = NULL;
-            return TRUE;
+            return true;
         }
     }
 
     // we couldn't store old stack or allocate a new one so we cannot provide new empty stack
-    return FALSE;
+    return false;
 }
 
-BOOL_T ddtrace_coms_buffer_data(uint32_t group_id, const char *data, size_t size) {
+bool ddtrace_coms_buffer_data(uint32_t group_id, const char *data, size_t size) {
     if (!data) {
-        return FALSE;
+        return false;
     }
 
-    if (data && size == 0) {
+    if (size == 0) {
         size = strlen(data);
+        if (size == 0) {
+            return false;
+        }
     }
 
-    if (size == 0) {
-        return FALSE;
-    }
-    BOOL_T store_result = store_data(group_id, data, size);
```

Comment:
This is where the potential bug was. `BOOL_T` is a `uint8_t` and `store_group` returns a `uint32_t`.